libclamav/matcher.c

```diff
@@ -802,6 +802,8 @@ cl_error_t cli_scan_desc(int desc, cli_ctx *ctx, cli_file_t ftype, uint8_t ftonl
     if (NULL != new_map) {
         funmap(new_map);
     }
+    // Clear the next-layer attributes so we don't accidentally apply them to subsequent layers.
+    ctx->next_layer_attributes = 0;
```

Comment:
Per discussion, it would be safer to convert this to a parameter and pass around as an argument (possibly burying it in a parameter struct if the argument list gets onerous).